pyvelox/pyvelox.h

```diff
@@ -105,10 +106,76 @@ inline velox::variant pyToVariant(const py::handle& obj) {
     return pyToVariant<velox::TypeKind::DOUBLE>(obj);
   } else if (py::isinstance<py::str>(obj)) {
     return pyToVariant<velox::TypeKind::VARCHAR>(obj);
+  } else if (py::isinstance<py::list>(obj)) {
+    py::list obj_as_list = py::cast<py::list>(obj);
+    std::vector<velox::variant> result;
+    for (auto& item : obj_as_list) {
+      result.push_back(pyToVariant(item));
+      if (result.front().kind() != result.back().kind()) {
+        throw py::type_error("Array must consist of elements of only one kind");
```

Comment:
what if its a null element ?

Reply:
Based on our discussion, Just adding these as comments in case it helps : 

: ARRAY<BIGINT>, ARRAY<ARRAY<BIGINT>>, ARRAY<T> ...., 
T -> Nulls 
 make_list({ 1 ,2 3 } -> vector<bigint>
 make_list({{1, 2}, {2, 3}} -> vector<array<bigint>>
 make_list{{}} -> vector<array<?>>, vector<map<?>> 
 make_list({{}}, type=array<int>) 
 
 //Infering
 { {} , None, { {}} , {{1}} } -> vector<array<array<int>>  
 
 // How Velox maps complex types: 
 
{ { 1, 2 }, { 3, 4, 5}, {Nil, 7}, {1 , 2}} -> Vector<ARRAY<INT>>

// Sometime in the future (maybe?)
 ARRAY VECTOR : ARRAY<T> : 
 -   Base Vector  : Vector <T> : Vector<int> -> { 1, 2, 3, 4,5, Nil, 7} : Nuls buffer : { 5th bit: set }, buffer of ints: 1, 2,... 
 -   Nulls buffer {}
 -  Lengths:  {2, 3, 2, 2}
  - Offsets : {0, 2, 5, 0 }